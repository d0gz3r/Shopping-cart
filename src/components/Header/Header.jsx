import { useSelector } from 'react-redux';

import { NavLink } from 'react-router-dom';
import basket from './img/shopping-cart.svg';
import styles from './Header.module.css';
import { useState, useEffect } from 'react';

const Header = () => {
  const [counter, setCounter] = useState(0);

  const storeData = useSelector(store => store.basketReducer);

  useEffect(() => {
    const arr = Object.entries(storeData);
    
    setCounter(arr.length);
  }, [storeData]);

  return (
    <div className={styles.container}>
      <ul className={styles.container__list}>
        <li className={styles.container__item}><NavLink to="/" exact>Главная</NavLink></li>
        <li className={styles.container__item}><NavLink to="/first" exact>Категория 1</NavLink></li> 
        <li className={styles.container__item}><NavLink to="/second" exact>Категория 2</NavLink></li> 
        <li className={styles.container__item}><NavLink to="/basket" exact><img className={styles.basket} src={basket} alt="" /><span className={styles.counter}>{counter}</span></NavLink></li> 
      </ul>
    </div>
  )
}


export default Header;