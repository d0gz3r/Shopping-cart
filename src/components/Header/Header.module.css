.link{
  color: #adb1b8;
  text-decoration: none;
  transition: .8s;
  font-family: 'Rubik', sans-serif;
  font-size: 16px;
  margin-right: 10px;
}
.nav-link-selected{
  color: red;
}
.link__basket{
  text-decoration: none;
}
.link:hover{
  color: #fff;
  text-decoration: underline;
}
.basket{
  width: 35px;
  height: 35px;
}
.counter{
  color: #adb1b8;
  font-size: 25px;
  font-weight: normal;
  margin-left: 10px;
}

@media (max-width: 991px){
  .basket{
    width: 25px;
    height: 25px;
  } 
  .counter{
    font-size: 20px;
  }
}