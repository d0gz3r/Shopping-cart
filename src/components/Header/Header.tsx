import React from 'react';

import { useTypedSelector } from '../../hooks/useTypedSelector';

import { Link } from 'react-router-dom';

// Bootstrap
import { Navbar, Container, Nav} from 'react-bootstrap';
//

import basket from './img/shopping-cart.svg';
import styles from './Header.module.css';
import { useState, useEffect } from 'react';

const Header: React.FC = () => {
  const [counter, setCounter] = useState<number>(0);

  const storeData = useTypedSelector(store => store.basket);

  useEffect(() => {
    const arr = Object.values(storeData);
    let sumOfGoods: number = 0;

    if(arr.length){
      arr.forEach((elem) => {
        sumOfGoods += elem.numberOfGoods;
      })
    }
   
    setCounter(sumOfGoods);
  }, [storeData]);


  return (
    <>
      <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
        <Container>
        <Navbar.Brand>STORE</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav data-testid="navbar" className="me-auto">
            <Nav.Link><Link data-testid="home-link" className={styles.link} to="/">Главная</Link></Nav.Link>
            <Nav.Link><Link className={styles.link} to="/first">Категория 1</Link></Nav.Link>
            <Nav.Link><Link className={styles.link} to="/second">Категория 2</Link></Nav.Link>
            <Nav.Link><Link data-testid="contact-link" className={styles.link} to="/map">Контакты</Link></Nav.Link>
          </Nav>
          <Nav>
            <Nav.Link><Link className={styles.link__basket} to="/basket"><img className={styles.basket} src={basket} alt="" /><span className={styles.counter}>{counter}</span></Link></Nav.Link>
          </Nav>
        </Navbar.Collapse>
        </Container>
      </Navbar>
    </>
  )
}


export default Header;