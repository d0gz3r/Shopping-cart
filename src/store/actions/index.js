import { ADD_GOOD_TO_BASKET, REMOVE_GOOD_FROM_BASKET } from "../constants/actionTypes"

export const addGoodToBasket = (good) => ({
  type: ADD_GOOD_TO_BASKET,
  payload: good
})

export const removeGoodFromBasket = (goodId) => ({
  type: REMOVE_GOOD_FROM_BASKET,
  payload: goodId
})