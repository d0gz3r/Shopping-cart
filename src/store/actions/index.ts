import { ADD_GOOD_TO_BASKET, REMOVE_GOOD_FROM_BASKET } from "../constants/actionTypes";
import { IInitialState } from "../../types/types";

export const addGoodToBasket = (good: IInitialState) => ({
  type: ADD_GOOD_TO_BASKET,
  payload: good
})

export const removeGoodFromBasket = (goodId: number) => ({
  type: REMOVE_GOOD_FROM_BASKET,
  payload: goodId
})