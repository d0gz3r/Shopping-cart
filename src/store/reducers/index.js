import basketReducer from "./basketReducer";
import { combineReducers } from "redux";

export default combineReducers({
  basketReducer
})