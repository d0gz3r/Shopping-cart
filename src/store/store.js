import { createStore } from "redux";
import roorReducer from './reducers';

const store = createStore(roorReducer);

export default store;