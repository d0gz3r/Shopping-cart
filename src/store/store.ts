import { createStore } from "redux";
import { rootReducer } from "./reducers";
import { setLocalStorage } from "../utils/localStorage";

const store = createStore(rootReducer);

store.subscribe(() => {
  setLocalStorage('store', store.getState().basket);
});

export default store;