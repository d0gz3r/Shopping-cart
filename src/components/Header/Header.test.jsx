import React from "react";
import Header from "./Header";
import { Provider } from "react-redux";
import store from '../../store/store';
import {shallow} from 'enzyme';

describe('Testing <Header/>', () => {
  it('Header have rendered', () => {
    const header = shallow(<Provider store={store}><Header/></Provider>);
    expect(header).toMatchSnapshot();
  })
});