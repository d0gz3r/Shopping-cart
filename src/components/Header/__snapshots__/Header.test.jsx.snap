// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Header/> Header have rendered 1`] = `ShallowWrapper {}`;
